features/netsocket/DNS.h

```diff
@@ -35,8 +35,25 @@ class DNS {
      *  @return         NSAPI_ERROR_OK on success, negative error code on failure.
      */
     virtual nsapi_error_t gethostbyname(const char *host,
-                                        SocketAddress *address, const char *interface_name, nsapi_version_t version = NSAPI_UNSPEC) = 0;
+                                        SocketAddress *address, nsapi_version_t version = NSAPI_UNSPEC) = 0;
 
+    /** Translate a hostname to an IP address with specific version using network interface name.
+     *
+     *  The hostname may be either a domain name or an IP address. If the
+     *  hostname is an IP address, no network transactions will be performed.
+     *
+     *  If no stack-specific DNS resolution is provided, the hostname
+     *  will be resolve using a UDP socket on the stack.
+     *
+     *  @param host     Hostname to resolve.
+     *  @param address  Pointer to a SocketAddress to store the result.
+     *  @param version  IP version of address to resolve, NSAPI_UNSPEC indicates
+     *  @param interface_name  Network interface_name
```

Comment:
There is also this version of the parameter in commit:
@param interface_name Network interface name
Change all to same format... maybe the "Network interface name" is better.

Reply:
Done